components/dashboard/src/user-context.tsx

```diff
@@ -5,18 +5,33 @@
  */
 
 import { User } from "@gitpod/gitpod-protocol";
+import { BillingMode } from "@gitpod/gitpod-protocol/lib/billing-mode";
 import React, { createContext, useState } from "react";
+import { getGitpodService } from "./service/service";
 
 const UserContext = createContext<{
     user?: User;
     setUser: React.Dispatch<User>;
+    userBillingMode?: BillingMode;
```

Comment:
I think this should be a `React.Dispatch`, in order for changes to make their way to all the context users (e.g. any page with settings menu):

```suggestion
    userBillingMode?: React.Dispatch<BillingMode>;
```